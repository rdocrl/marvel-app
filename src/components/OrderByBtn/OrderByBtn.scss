.order-by-btn {
  background: red;
  padding: 10px;
  border: none;
  border-radius: 6px;
  color: #fff;
  font-weight: bold;
  cursor: pointer;
}
