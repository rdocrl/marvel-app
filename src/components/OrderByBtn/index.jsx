import './OrderByBtn.scss';

const OrderByBtn = ({ onClick, currentOrder }) => {
  return (
    <button className="order-by-btn" onClick={onClick}>
      Order By {currentOrder ? (currentOrder === '+' ? '▲' : '▼') : ''}
    </button>
  );
};

export default OrderByBtn;
