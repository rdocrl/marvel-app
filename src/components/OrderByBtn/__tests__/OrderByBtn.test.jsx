import { render } from '@testing-library/react';
import OrderByBtn from '../';

describe('<OrderByBtn /> tests', () => {
  describe('Snapshots', () => {
    it('Should render', () => {
      const { asFragment } = render(<OrderByBtn />);
      expect(asFragment()).toMatchSnapshot();
    });

    it('Should render with current ascending order provided', () => {
      const { asFragment } = render(<OrderByBtn currentOrder={'+'} />);
      expect(asFragment()).toMatchSnapshot();
    });

    it('Should render with current descending order provided', () => {
      const { asFragment } = render(<OrderByBtn currentOrder={'-'} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
