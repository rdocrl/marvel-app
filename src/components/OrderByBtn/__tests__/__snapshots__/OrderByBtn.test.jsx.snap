// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderByBtn /> tests Snapshots Should render 1`] = `
<DocumentFragment>
  <button
    class="order-by-btn"
  >
    Order By 
  </button>
</DocumentFragment>
`;

exports[`<OrderByBtn /> tests Snapshots Should render with current ascending order provided 1`] = `
<DocumentFragment>
  <button
    class="order-by-btn"
  >
    Order By ▲
  </button>
</DocumentFragment>
`;

exports[`<OrderByBtn /> tests Snapshots Should render with current descending order provided 1`] = `
<DocumentFragment>
  <button
    class="order-by-btn"
  >
    Order By ▼
  </button>
</DocumentFragment>
`;
